pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,409 @@ import (
+		table.Entry("should unarchive tar", baseTestImage, false, []string{image.ExtTar}),
+		table.Entry("should unpack .tar.gz", baseTestImage, false, []string{image.ExtTar, image.ExtGz}),
+		// Disabled until issue 335 is resolved
+		// https://github.com/kubevirt/containerized-data-importer/issues/335
+		//table.Entry("should unpack .tar.xz", baseTestImage, false, []string{image.ExtTar, image.ExtXz}),
+		table.Entry("should convert .qcow2", baseTestImage, true, []string{image.ExtQcow2}),
+		table.Entry("should convert and unpack .qcow2.gz", baseTestImage, false, []string{image.ExtQcow2, image.ExtGz}),
+		table.Entry("should convert and unpack .qcow2.xz", baseTestImage, false, []string{image.ExtQcow2, image.ExtXz}),
+		table.Entry("should convert and untar .qcow2.tar", baseTestImage, false, []string{image.ExtQcow2, image.ExtTar}),
+		table.Entry("should convert and untar and unpack .qcow2.tar.gz", baseTestImage, false, []string{image.ExtQcow2, image.ExtTar, image.ExtGz}),
+		table.Entry("should convert and untar and unpack .qcow2.tar.xz", baseTestImage, false, []string{image.ExtQcow2, image.ExtTar, image.ExtXz}),
+		table.Entry("should pass through unformatted data", baseTestImage, false, []string{""}),
+	)
+})
+
+func getImageVirtualSize(outFile string) int64 {
+	//call qemu-img info
+	virtSizeParseLen := 8
+
+	//create command
+	cmd := fmt.Sprintf("qemu-img info %s | grep 'virtual size:'", outFile)
+	out, err := exec.Command("sh", "-c", cmd).Output()
+	if err != nil {
+		return 0
 	}
+	sOut := string(out)
 
-	if !started {
-		server.Shutdown(nil)
-		return nil, errors.New("Couldn't start http server")
-	}
+	index1 := strings.Index(sOut, "(")
+	sSize := sOut[index1+1 : len(sOut)-virtSizeParseLen]
 
-	return server, nil
+	vSize, err := strconv.ParseInt(sSize, 10, 64)
+	if err != nil {
+		return 0
+	}
+	return vSize
 }
 
-func TestNewDataStream(t *testing.T) {
-	type args struct {
-		endpt  string
-		accKey string
-		secKey string
-	}
+func testDir(parent string) string {
+	suf := util.RandAlphaNum(12)
+	return filepath.Join(parent, fmt.Sprintf(".tmp-%s", string(suf)))
+}
 
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	localImageBase := filepath.Join("file://", imageDir)
-
-	tests := []struct {
-		name    string
-		args    args
-		want    *DataStream
-		wantErr bool
-	}{
-		{
-			name:    "expect new DataStream to succeed with matching buffers",
-			args:    args{filepath.Join(localImageBase, "cirros-qcow2.img"), "", ""},
-			want:    createDataStream(filepath.Join(localImageBase, "cirros-qcow2.img"), "", ""),
-			wantErr: false,
-		},
-		{
-			name:    "expect new DataStream to fail with unsupported file type",
-			args:    args{filepath.Join(localImageBase, "badimage.iso"), "", ""},
-			want:    nil,
-			wantErr: true,
-		},
-		{
-			name:    "expect new DataStream to fail with invalid or missing endpoint",
-			args:    args{"", "", ""},
-			want:    nil,
-			wantErr: true,
-		},
+func emptyBuffer(ds *DataStream) {
+	buf := make([]byte, 32*1024)
```

Comment:
If the point of this func is to reach eof on all readers then I think you need a larger buf. Otherwise, I am not sure why you are reading 32k worth of the image? A comment would be helpful.

Reply:
See the for loop below here? It runs until the readers are exhausted, I am not actually interested in the content.

Reply:
Yes I see that. But what is the point in "exhausting" the readers and what does that actually mean. I was thinking you wanted to reach EOF but maybe it is something else. Is the hard-coded 32k buf a magic size (ie related to the biggest buf a reader could have? I assume this func is needed so you can call NewDataStream more than once and have a fresh image to process each time. There is the concept of _resetting_ a file which basically moves the current record ptr to 0. Is this the idea here?

Reply:
Hmm okay, well I must have been doing something wrong in my earlier experiments because the httptest server would just hang waiting for something to read the stream, so I had to explicitly read the stream to make it work. But I since switched to the file server and it is no longer needed.

But originally it was just to simply read the stream so the server wouldn't lock waiting for something to read the stream.

Reply:
@awels I'm not saying you are doing something wrong here. More that I don't understand it...